src/model/messages/mrtd/EMrtdDataSubmitMessage.ts

```diff
@@ -17,13 +59,77 @@ export class EMrtdDataSubmitMessage extends BaseMessage {
       this.threadId = options.threadId
       this.timestamp = options.timestamp ?? new Date()
       this.connectionId = options.connectionId
-      this.dataGroups = options.dataGroups
+      this.dataGroups = this.parseDataGroups(options.dataGroups)
     }
   }
 
   public readonly type = EMrtdDataSubmitMessage.type
   public static readonly type = MessageType.EMrtdDataSubmitMessage
 
   @Expose()
-  public dataGroups!: EMrtdData
+  public dataGroups!: EMrtdRawData
+
+  public parseDataGroups({ raw, parsed }: EMrtdData): EMrtdRawData {
+    if (!parsed || !parsed.fields) return this.dataGroups
+
+    const dataUrls = parsed.fields.images.map(image => {
+      const faceMimeType = image.imageType === 1 ? 'image/jp2' : 'image/jpeg'
+      return `data:${faceMimeType};base64,${image.imageData.toString('base64')}`
+    })
+
+    const mrzString = parsed.fields.mrzData
+    const formattedMrz = (len => {
+      switch (len) {
+        case 88:
+          return mrzString.slice(0, 44) + '\n' + mrzString.slice(44)
+        case 72:
+          return mrzString.slice(0, 36) + '\n' + mrzString.slice(36)
+        case 90:
+          return mrzString.slice(0, 30) + '\n' + mrzString.slice(30, 60) + '\n' + mrzString.slice(60)
+        case 30:
+          return mrzString
+        case 18:
+          return mrzString.slice(0, 9) + '\n' + mrzString.slice(9)
+        default:
+          throw new Error(`Unrecognized MRZ length: ${len}`)
+      }
+    })(mrzString.length)
+    const parsedMrz = Mrz.parse(formattedMrz)
+
+    const newEmrtdData: EMrtdRawData = {
+      raw: raw,
+      processed: {
+        ef_dg1: {
```

Comment:
There is no need to put fields within specific objects for each EF. It is simpler for the clients to have a plain object containing all processed, intepreted fields without the need of dealing with something that is very specific to ICAO 9303 spec.

This means that the structure of the resulting object would be:
```
{
raw: { COM: 'xxx', DG1: 'yyy', ... },
processed: {
      documentType: parsedMrz.format,
          documentNumber: parsedMrz.documentNumber,
          issuingState: parsedMrz.fields.issuingState,
          dateOfExpiry: parsedMrz.fields.expirationDate,  // TODO: Check and specify date format
          sex: parsedMrz.fields.sex,
          nationality: parsedMrz.fields.nationality,
          lastName: parsedMrz.fields.lastName,
          firstName: parsedMrz.fields.firstName,
          mrzOptionalData: parsedMrz.fields.optional1,
          faceImages: dataUrls,
          nameOfHolder: parsed.fields.additionalPersonalData?.nameOfHolder ?? `${parsedMrz.fields.lastName} ${firstName: parsedMrz.fields.firstName}`,
          dateOfBirth: parsed.fields.additionalPersonalData?.fullDateOfBirth ?? parsedMrz.fields.birthDate, // TODO: Check and specify date format
          otherNames: parsed.fields.additionalPersonalData?.otherNames,
          personalNumber: parsed.fields.additionalPersonalData?.personalNumber,
          placeOfBirth: parsed.fields.additionalPersonalData?.placeOfBirth,
          permanentAddress: parsed.fields.additionalPersonalData?.permanentAddress,
          telephone: parsed.fields.additionalPersonalData?.telephone,
          profession: parsed.fields.additionalPersonalData?.profession,
          title: parsed.fields.additionalPersonalData?.title,
          personalSummary: parsed.fields.additionalPersonalData?.personalSummary,
          custodyInformation: parsed.fields.additionalPersonalData?.custodyInformation,
}
```

Reply:
fixed, but I'm still unsure about the format to use. The passports maintain the YYMMDD format, so for now, I believe that should be fine